h2/src/main/org/h2/expression/function/ToChar.java

```diff
@@ -828,9 +828,32 @@ public static String toCharDateTime(Session session, Value value, String format,
             } else if (containsAt(format, i, "TZD") != null) {
                 output.append(getTimeZone(session, value, true));
                 i += 3;
+            } else if (containsAt(format, i, "TZH") != null) {
+                String result;
+                int hours = DateTimeFunctions.extractIntegerField(session, value, DateTimeFunctions.TIMEZONE_HOUR);
+                StringBuilder b = new StringBuilder();
+                b.append( hours < 0 ? '-' : '+');
+                hours = Math.abs(hours);
+                if(hours == 0) {
+                    result = b.append("00").toString();
+                } else {
+                    result = StringUtils.appendTwoDigits(b, hours).toString();
+                }
+                output.append(result);
+                i += 3;
 
-                // Week
+            } else if (containsAt(format, i, "TZM") != null) {
+                String result;
+                int mins = Math.abs(DateTimeFunctions.extractIntegerField(session, value, DateTimeFunctions.TIMEZONE_MINUTE));
+                if(mins == 0) {
+                    result = "00";
+                } else {
+                    result = StringUtils.appendTwoDigits(new StringBuilder(), mins).toString();
+                }
```

Comment:
`output.append("00")` / `StringUtils.appendTwoDigits(output, mins)`.

Also add a space between `if` and `(` here and above.